### PR TITLE
Editor for polls - Fix for large file upload

### PR DESCRIPTION
**Pull request by DevPhilB**
**Features:**
- Polls with multiple answers can now be created
- Answers at each position can be deleted
- Check if poll has two answers as a minimum
- API for polls (create, get=votes, update=vote and delete)
- Radio buttons can be resetted
- Users can vote
- Users can see the result
- Result is displayed as bar chart with vote number/percent

**Fixes:**
- Upload of large files (>1 MB) doesn't work (#236)
- Package names conflict with existing npm-packages

Poll-Create
Poll-Vote
Poll-Result

### CODE REVIEW

File: db/HelperQueries/quickSetup.pgsql [hunk 1]
```diff
@@ -33,6 +33,21 @@ CREATE TABLE public.post (
     interactive_due_date bigint
 );
 
+CREATE TABLE public.poll (
+    poll_id serial NOT NULL PRIMARY KEY,
```
Comment (MistereoSC) [inline db/HelperQueries/quickSetup.pgsql]: Primary key should be named 'id'

File: db/HelperQueries/quickSetup.pgsql [hunk 2]
```diff
@@ -33,6 +33,21 @@ CREATE TABLE public.post (
...
+    post_id integer REFERENCES post (post_id) ON DELETE CASCADE,
+    answer_id integer NOT NULL,
+    vote integer NOT NULL
+);
+
+CREATE TABLE public.survey (
+    survey_id serial NOT NULL PRIMARY KEY,
```
Comment (MistereoSC) [inline db/HelperQueries/quickSetup.pgsql]: Primary key should be named 'id'

File: db/HelperQueries/quickSetup.pgsql [hunk 3]
```diff
@@ -33,6 +33,21 @@ CREATE TABLE public.post (
     interactive_due_date bigint
 );
 
+CREATE TABLE public.poll (
+    poll_id serial NOT NULL PRIMARY KEY,
+    post_id integer REFERENCES post (post_id) ON DELETE CASCADE,
```
Comment (MistereoSC) [inline db/HelperQueries/quickSetup.pgsql]: foreign key should reference to post (id)

File: db/HelperQueries/quickSetup.pgsql [hunk 4]
```diff
@@ -33,6 +33,21 @@ CREATE TABLE public.post (
...
+    answer_id integer NOT NULL,
+    vote integer NOT NULL
+);
+
+CREATE TABLE public.survey (
+    survey_id serial NOT NULL PRIMARY KEY,
+    post_id integer REFERENCES post (post_id) ON DELETE CASCADE,
```
Comment (MistereoSC) [inline db/HelperQueries/quickSetup.pgsql]: foreign key should reference to post (id)

File: db/survey.pgsql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE public.survey (
+    id serial NOT NULL PRIMARY KEY,
+    post_id integer REFERENCES post (post_id) ON DELETE CASCADE,
```
Comment (MistereoSC) [inline db/survey.pgsql]: foreign key should reference to post (id)

File: sails/api/controllers/polls/create.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    missingParams: {
+      description: 'Missing parameters',
+      statusCode: 400
+    },
+    invalidParams: {
+      description: 'Invalid parameters',
+      statusCode: 400
+    },
+    authRequired: {
```
Comment (MistereoSC) [inline sails/api/controllers/polls/create.js]: Unused exit code. Authentication is checked via sails policies

File: sails/config/custom.js [hunk 7]
```diff
@@ -35,8 +35,8 @@ module.exports.custom = {
      **************************************************************************/
     
     // Allows letters and numbers only
-    BOARDNAME_REGEX: '^[a-zA-Z0-9äÄöÖüÜß \'\-\.\,]{2,50}$',
-    POSTTITLE_REGEX: '^[a-zA-Z0-9äÄöÖüÜß \'\-\.\,]{2,50}$',
+    BOARDNAME_REGEX: '^[a-zA-Z0-9äÄöÖüÜß \'\-\.\,\/\&]{2,50}$',
+    POSTTITLE_REGEX: '^[a-zA-Z0-9äÄöÖüÜß \_]{2,49}[?]?$',
```
Comment (MistereoSC) [inline sails/config/custom.js]: Special characters    '  -  ,  .   should still be allowed within post titles

File: sails/api/controllers/polls/delete.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
```
Comment (MistereoSC) [inline sails/api/controllers/polls/delete.js]: Redundant. Poll should be automatically deleted via posts/delete.js, or inside DB on cascade

File: sails/api/models/Poll.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  tableName: 'poll',
+  primaryKey: 'id',
+  attributes: {
+    id: { type: 'number', autoIncrement: true, unique: true, columnName: 'id' },
+    postId: { type: 'number', columnName: 'post_id' },
+    answerId: { type: 'number', columnName: 'answer_id' },
+    votes: { type: 'number', columnName: 'vote' }
```
Comment (MistereoSC) [inline sails/api/models/Poll.js]: Maybe add defaultsTo: 0 , to prevent null collumn

File: sails/test/poll.test.js [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+      });
+  }, 1000);
+});
+
+const deletePollData = {
+  postId: postId
+};
+
+// Delete the poll
+test('POLL::: Delete poll', () => {
```
Comment (MistereoSC) [inline sails/test/poll.test.js]: Poll should be automatically deleted when the post gets deleted

**Comment by DevPhilB** [comment]
> 
> 
> Everything works and looks good. Especially the bar chart representation looks professional.
> All custom tests passed successfully.
> 
> Only ESLint shows the following error: "Expected linebreaks to be 'LF' but found 'CRLF'".

Please approve this PR.

ESLint doesn't throw any error on our [CI](https://circleci.com/gh/conclurer/midnightboard/328), it's a local thing and you can change it in Visual Studio Code (bottom right corner) from CRLF to LF if you want. Git does this automatically for you if you commit your changes.

**Pull request merged**